tests/fetch.spec.ts

```diff
@@ -0,0 +1,255 @@
+import { CacheManager } from '../src/cacheManager.ts';
+import { it, expect, beforeEach, vi } from 'vitest';
+import { FetchService } from '../src/fetchService.ts';
+import type { CacheConfig } from '../src/types/fetchService.ts';
+import { ContentType } from '../src/constants.ts';
+import { defaults } from 'make-fetch-happen';
+
+// Mock make-fetch-happen
+vi.mock('make-fetch-happen', () => ({
+  defaults: vi.fn().mockImplementation(() => vi.fn()),
+}));
+
+// Mock CacheManager
+vi.mock('./cacheManager');
+
+const mockConfig: CacheConfig = {
+  basePath: '/tmp/cache',
+  contentTypes: {
+    [ContentType.WEB_PAGE]: {
+      path: 'web-pages',
+      maxAge: 3600000,
+      cacheMode: 'force-cache',
+      retries: 3,
+    },
+    [ContentType.MARKDOWN]: {
+      path: 'search-indexes',
+      maxAge: 7200000,
+      cacheMode: 'default',
+      retries: 2,
+    },
+  },
+};
+
+// Setup CacheManager mock with proper methods
+mockCacheManager = {
+  clearCache: jest.fn().mockResolvedValue(undefined),
```

Comment:
Should these not be `vi.fn()` rather than `jest.fn()`?

Reply:
Haven't gone through any of the test files that end with `*.spec.ts` - but yes